mgr/dashboard: Logs Page E2E Tests

adk3798 opened the pull request:
Tests that logs page creates audit logs in response to pool creation/deletion
Tests that logs page creates audit logs in response to configuration changes
Alter configuration po to put config name in filter box so you can edit configs not on first page (first 10)

Fixes: https://tracker.ceph.com/issues/41035

Signed-off-by: Adam King <adking@redhat.com>
Signed-off-by: Rafael Quintero <rquinter@redhat.com>

adk3798 commented:
jenkins test dashboard

adk3798 commented:
jenkins test dashboard

tspmelo commented on src/pybind/mgr/dashboard/frontend/e2e/cluster/logs.po.ts:
```diff
@@ -1,8 +1,130 @@
+      .first()
+      .sendKeys(configname);
+
+    const audit_logs_tab = element(by.cssContainingText('.tab-pane.active', ''));
```
```suggestion
    const audit_logs_tab = element(by.css('.tab-pane.active'));
```

tspmelo commented on src/pybind/mgr/dashboard/frontend/e2e/cluster/logs.po.ts:
```diff
@@ -1,8 +1,130 @@
+      .sendKeys(configname);
+
+    const audit_logs_tab = element(by.cssContainingText('.tab-pane.active', ''));
+    const audit_logs_body = audit_logs_tab.element(by.cssContainingText('.card-body', ''));
```
```suggestion
    const audit_logs_body = audit_logs_tab.element(by.css('.card-body'));
```

tspmelo commented on src/pybind/mgr/dashboard/frontend/e2e/cluster/logs.e2e-spec.ts:
```diff
@@ -32,4 +45,49 @@ describe('Logs page', () => {
+
+  describe('audit logs respond to editing configuration setting test', () => {
+    beforeAll(() => {
+      logs.navigateTo();
```
```suggestion
```

tspmelo commented on src/pybind/mgr/dashboard/frontend/e2e/cluster/logs.po.ts:
```diff
@@ -1,8 +1,130 @@
+      .first()
+      .sendKeys(poolname);
+
+    const audit_logs_tab = element(by.cssContainingText('.tab-pane.active', ''));
```
```suggestion
    const audit_logs_tab = element(by.css('.tab-pane.active'));
```

tspmelo commented on src/pybind/mgr/dashboard/frontend/e2e/cluster/logs.po.ts:
```diff
@@ -1,8 +1,130 @@
+      .sendKeys(poolname);
+
+    const audit_logs_tab = element(by.cssContainingText('.tab-pane.active', ''));
+    const audit_logs_body = audit_logs_tab.element(by.cssContainingText('.card-body', ''));
```
```suggestion
    const audit_logs_body = audit_logs_tab.element(by.css('.card-body'));
```

adk3798 commented:
Updates:
 - Changed some lines in logs po that used cssContainingText with no text to use css or just $
 - Changed lines getting filter box in configuration po to use $ instead of $$
 - Removed beforeAll block in new logs e2e Describe blocks since they were just calling navigateTo which is done within all It blocks within those Describe blocks
